agent/exec/container/adapter.go

```diff
@@ -61,15 +61,21 @@ func (c *containerAdapter) pullImage(ctx context.Context) error {
 
 	dec := json.NewDecoder(rc)
 	m := map[string]interface{}{}
+	spamLimiter := time.NewTicker(time.Millisecond * 500).C
```

Comment:
I'm adding that defer anyway, but would this leak or would the ticker get GC'd after the function ended anyway?